test/conformance/api/v1/service_test.go

```diff
@@ -581,3 +582,109 @@ func TestAnnotationPropagation(t *testing.T) {
 		t.Error("Route still has `juicy` annotation")
 	}
 }
+
+// TestServiceCreateAndUpdateWithMultipleContainers tests both Creation and Update paths for a service. The test performs a series of Update/Validate steps to ensure that
```

Comment:
I think this test is testing a lot of random stuff, that we probably can avoid.
I'd actually just setup configuration and verify the pod has come up and ready and has 3 containers (+qp).

Reply:
Yep that make sense

Updated according to comment please do have a look

Thank you